Added .destroy() function with tests for File-Bin

acareaga opened the pull request:
Implemented the destroy (aka delete.. ?) for file-bin. 

Please let me know if you see any errors or areas that should be improved

stevekinney committed:
Swap out __dirname for process.cwd()

Wrote tests for destroy function. Added to prototype

Need to fix destroy, unlink might not be the best method

Added destroy method for the filesystem with tests

Removed package.json from commit

Destroy for filebin is implemented with tests. Fixed catch(done) on test

Removed mocha as a dependency sinces its already in the package.json as dev dep

acareaga committed:
Wrote tests for destroy function. Added to prototype

acareaga committed:
Need to fix destroy, unlink might not be the best method

acareaga committed:
Added destroy method for the filesystem with tests

acareaga committed:
Removed package.json from commit

acareaga committed:
Destroy for filebin is implemented with tests. Fixed catch(done) on test

acareaga committed:
Removed mocha as a dependency sinces its already in the package.json as dev dep

stevekinney commented on index.js:
```diff
@@ -57,6 +57,15 @@ FileBin.prototype.write = function (fileName, data) {
+  return new RSVP.Promise((resolve, reject) => {
+    fs.unlink(path.join(this.base, fileName), (error) => {
+      if (error) { return reject(error); }
+      resolve(formatFile(fileName));
```
It's kind of weird to return our now dead file. What about `resolve(true)` instead?

stevekinney commented:
Can you add some documentation to the README and squash the commits?

rossedfort committed:
add rename method, tests, and update README.md

stevekinney committed:
Merge pull request #8 from rossedfort/master

Add rename method

stevekinney commented:
We also have some conflicts now.

acareaga commented:
Sounds good - I'll fix those conflicts, clean up the commits, and add some documentation tonight. 

Resolve true makes more sense. I'm still a little confused with the promise structure.

acareaga committed:
Updated the destroy method to resolve true instead of the expired path

acareaga committed:
Updated the destroy method to resolve true instead of the expired path

acareaga committed:
Added documentation for the destroy method, fixed the resolve to true, and squashed unnecessary commits

acareaga committed:
Added destroy function, documentation, and tests. Fixed merge conflity

acareaga committed:
Fixed merge conflict

acareaga commented:
I have to be missing something... tried for over an hour to squash previous commits. Git says I picked and squashed the correct files and rebased to the most recent commit locally but the PR still shows 10 commits.

stevekinney commented:
Force push?

> On Feb 23, 2016, at 8:38 PM, Aaron Careaga notifications@github.com wrote:
> 
> I have to be missing something... tried for over an hour to squash previous commits. Git says I picked and squashed the correct files and rebased to the most recent commit locally but the PR still shows 10 commits.
> 
> —
> Reply to this email directly or view it on GitHub.

acareaga committed:
Added destroy method with tests and documentation

acareaga committed:
Merge branch 'master' of github.com:acareaga/file-bin
